src/jit/scopeinfo.cpp

```diff
@@ -154,6 +154,92 @@ CodeGenInterface::siVarLoc::siVarLoc(const LclVarDsc* varDsc, regNumber baseReg,
     }
 }
 
+//------------------------------------------------------------------------
+// Equals: Compares first reference and then values of the structures.
+//
+// Arguments:
+//    lhs   - a "siVarLoc *" to compare.
+//    rhs   - a "siVarLoc *" to compare.
+//
+// Notes:
+//    Return true if both are nullptr.
+//
+// static
+bool CodeGenInterface::siVarLoc::Equals(const siVarLoc* lhs, const siVarLoc* rhs)
+{
+    if (lhs == rhs)
+    {
+        // Are both nullptr or the same reference
+        return true;
+    }
+    if ((nullptr == lhs) || (nullptr == rhs))
+    {
+        // Just one of them is a nullptr
+        return false;
+    }
+
+    bool areEquals = true;
+
+    if (lhs->vlType == rhs->vlType)
+    {
+        // If neither is nullptr, and are not the same reference, compare values
+        switch (lhs->vlType)
+        {
+            case VLT_STK:
+            case VLT_STK_BYREF:
+                areEquals &= lhs->vlStk.vlsBaseReg == rhs->vlStk.vlsBaseReg;
```

Comment:
all these also should be parentheses.
`areEquals &= (lhs->vlStk.vlsBaseReg == rhs->vlStk.vlsBaseReg);`.

and I would change it to `bool areEquals;` and change each first `&=` to `=` in cases.

Then in `case VLT_COUNT:` do `areEquals = true;` and delete the comment.

In such case compiler will check that you do not have paths where `areEquals` is not initialized and `case VLT_COUNT:` won't be that tricky.

But maybe the second part is just my taste and you can ignore it, it is better to ask @BruceForstall

Reply:
Just make it `return (a == b) && (c == d);`

Reply:
Agree, I vote for:
```
case VLT_REG_STK:
return (lhs->vlRegStk.vlrsReg == rhs->vlRegStk.vlrsReg) && (lhs->vlRegStk.vlrsStk.vlrssBaseReg == rhs->vlRegStk.vlrsStk.vlrssBaseReg) && (lhs->vlRegStk.vlrsStk.vlrssOffset == rhs->vlRegStk.vlrsStk.vlrssOffset);
```
with jit-format new lines.

It also allows you not to mix logical and bit operations (that is pretty safe for bool but it is not a good thing).

Reply:
I have applied your suggestions. I don't think this has made things easier to read.